### PR TITLE
Add UG for Sort and Filter

### PR DESCRIPTION
**Pull request by ivanleekk**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -253,20 +253,47 @@ Examples:
...
+Filters the applicant list based on fields given by the user.
+
+Format: `filter [n/Name] [e/Email] [p/Position] [hp/Phone] [s/Status] [lts/Score] [gts/Score]`
+* All fields are optional, however at least one of the optional fields must be provided.
+* Any combination of multiple different fields is allowed. 
+* Only one of each field can be provided.
+* Only applicants matching all fields will be returned. (i.e. AND search).
+* for `[n/Name]` field, only applicants whose name contains the full substring will be returned, e.g. `n/Ivan Chew` will **NOT** return `Ivan Lee`.
+* `[n/Name]` `[e/Email]` `[p/Position]` fields are case-insensitive, e.g. `n/JOHN` will return `john`.
+* `[lts/Score]` `[gts/Score]` fields include equality in filters, e.g. `gts/7` will return all applicants with a greater than or equal to `7`.
```
Comment (CelestineTan03) [inline docs/UserGuide.md:285]: It seems like the current implementation does not include equality 😮 Perhaps possible to add it as a future implementation 😎 but better to describe the current implementation here and exclude the part on equality 😄

**Review by CelestineTan03 — CHANGES_REQUESTED** [review]
Clear user guide for sort and filter, just a minor error regarding the `filter gts/Score` and `lts/Score` portion 😊

File: docs/UserGuide.md (hunk 1 above)
Comment (ivanleekk) [reply docs/UserGuide.md]: Seems like a bug, as the implementation was supposed to handle equality 🥸. But will fix that next week! For now I have changed the UG to reflect not equality!

**Review by craigtonlian — APPROVED** [review]
Lgtm! 😄

**Review by CelestineTan03 — APPROVED** [review]
Lgtm 👍

**Pull request merged**